File: scripts/gen_expired_trial_test.py
from reportlab.lib.pagesizes import letter
from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, Table, TableStyle, HRFlowable
from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
from reportlab.lib import colors
from reportlab.lib.units import inch

OUT = r"C:\Users\wallyd1\REDLINE\scripts\Test_Expired_Trial_Gate.pdf"

doc = SimpleDocTemplate(OUT, pagesize=letter,
    leftMargin=0.85*inch, rightMargin=0.85*inch,
    topMargin=0.85*inch, bottomMargin=0.85*inch)

styles = getSampleStyleSheet()
RED    = colors.HexColor('#cc0000')
DARK   = colors.HexColor('#111111')
MUTED  = colors.HexColor('#555555')
GREEN  = colors.HexColor('#1a7a1a')
BGRED  = colors.HexColor('#fff0f0')
BGGREEN= colors.HexColor('#f0fff4')
BGGRAY = colors.HexColor('#f5f5f5')

title_style = ParagraphStyle('T', parent=styles['Title'],
    textColor=RED, fontSize=20, spaceAfter=4, leading=24)
sub_style   = ParagraphStyle('S', parent=styles['Normal'],
    textColor=MUTED, fontSize=10, spaceAfter=16)
h2_style    = ParagraphStyle('H2', parent=styles['Heading2'],
    textColor=DARK, fontSize=13, spaceBefore=18, spaceAfter=6,
    borderPad=0, leading=16)
body_style  = ParagraphStyle('B', parent=styles['Normal'],
    textColor=DARK, fontSize=10, leading=15, spaceAfter=5)
code_style  = ParagraphStyle('C', parent=styles['Normal'],
    fontName='Courier', fontSize=8.5, textColor=colors.HexColor('#222'),
    backColor=BGGRAY, borderPad=6, leading=13, spaceAfter=6)
check_style = ParagraphStyle('CK', parent=styles['Normal'],
    fontSize=10, leading=17, textColor=DARK)
note_style  = ParagraphStyle('N', parent=styles['Normal'],
    fontSize=9, textColor=MUTED, leading=13)

story = []

# ── Header ──────────────────────────────────────────────────────
story.append(Paragraph("Expired Trial Plan Gate", title_style))
story.append(Paragraph("Test Procedure", ParagraphStyle('TS', parent=title_style, fontSize=15, textColor=DARK, spaceAfter=2)))
story.append(Paragraph("Version 1.0 &nbsp;|&nbsp; Date: 2026-06-17 &nbsp;|&nbsp; Product: Redlined1", sub_style))
story.append(HRFlowable(width="100%", thickness=2, color=RED, spaceAfter=14))

# ── Objective ───────────────────────────────────────────────────
story.append(Paragraph("Objective", h2_style))
story.append(Paragraph(
    "Verify that after a 7-day trial expires the plan gate locks all modules except "
    "<b>Dashboard</b>, <b>Settings</b>, and <b>Plans &amp; Gates</b>. "
    "Confirm that an active trial account still has unrestricted full access.",
    body_style))

# ── Pre-flight ──────────────────────────────────────────────────
story.append(Paragraph("Pre-Flight Checklist", h2_style))
pre = [
    "You have access to the Supabase Dashboard SQL Editor",
    "You have a test email address ready (e.g. testexpired@redlined1.com)",
    "redlined1.com is deployed and accessible in a browser",
    "You are NOT logged in as your main owner account during these tests",
]
for item in pre:
    story.append(Paragraph(f"&#9744; &nbsp; {item}", check_style))
story.append(Spacer(1, 8))

# ── Step 1 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 1 — Create the Test Account (one-time setup)", h2_style))
story.append(Paragraph("<b>1a.</b> Go to Supabase Dashboard → Authentication → Users → <b>Invite User</b>", body_style))
story.append(Paragraph("<b>1b.</b> Enter email: <font name='Courier'>testexpired@redlined1.com</font> and a test password", body_style))
story.append(Paragraph("<b>1c.</b> Confirm the profile row exists — run in SQL Editor:", body_style))
story.append(Paragraph(
    "SELECT id, plan, trial_ends_at FROM public.profiles\n"
    "WHERE id = (SELECT id FROM auth.users\n"
    "            WHERE email = 'testexpired@redlined1.com');",
    code_style))
story.append(Paragraph("<b>Expected:</b> row exists with plan = 'trial' and trial_ends_at ~7 days from signup.", note_style))

# ── Step 2 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 2 — Expire the Trial", h2_style))
story.append(Paragraph("Run this SQL in Supabase → SQL Editor:", body_style))
story.append(Paragraph(
    "UPDATE public.profiles\n"
    "SET plan = 'trial',\n"
    "    trial_ends_at = now() - INTERVAL '1 day'\n"
    "WHERE id = (\n"
    "  SELECT id FROM auth.users\n"
    "  WHERE email = 'testexpired@redlined1.com'\n"
    ");",
    code_style))
story.append(Paragraph("Then verify the change:", body_style))
story.append(Paragraph(
    "SELECT u.email, p.plan, p.trial_ends_at\n"
    "FROM auth.users u\n"
    "JOIN public.profiles p ON p.id = u.id\n"
    "WHERE u.email = 'testexpired@redlined1.com';",
    code_style))
story.append(Paragraph("<b>Expected:</b> trial_ends_at is in the past (yesterday).", note_style))

# ── Step 3 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 3 — Log In as the Expired Test Account", h2_style))
story.append(Paragraph("<b>3a.</b> Open a <b>private / incognito</b> browser window", body_style))
story.append(Paragraph("<b>3b.</b> Navigate to <font name='Courier'>redlined1.com</font> → Sign In", body_style))
story.append(Paragraph("<b>3c.</b> Enter credentials for <font name='Courier'>testexpired@redlined1.com</font>", body_style))

# ── Step 4 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 4 — Verify Module Lock-Down", h2_style))
story.append(Paragraph("Click every sidebar item and mark the result below.", body_style))
story.append(Spacer(1, 6))

# Accessible table
story.append(Paragraph("SHOULD BE ACCESSIBLE (3 modules):", ParagraphStyle('GL', parent=body_style, textColor=GREEN, fontName='Helvetica-Bold', spaceAfter=4)))
ok_rows = [
    ["Module", "Expected Result", "Pass?"],
    ["Dashboard",    "Opens normally — data is visible",        "[ ]"],
    ["Settings",     "Opens normally — can edit branding",      "[ ]"],
    ["Plans & Gates","Opens upgrade / subscription page",       "[ ]"],
]
ok_table = Table(ok_rows, colWidths=[1.7*inch, 3.8*inch, 0.7*inch])
ok_table.setStyle(TableStyle([
    ('BACKGROUND',  (0,0), (-1,0), GREEN),
    ('TEXTCOLOR',   (0,0), (-1,0), colors.white),
    ('FONTNAME',    (0,0), (-1,0), 'Helvetica-Bold'),
    ('FONTSIZE',    (0,0), (-1,-1), 9),
    ('ROWBACKGROUNDS', (0,1), (-1,-1), [BGGREEN, colors.white]),
    ('GRID',        (0,0), (-1,-1), 0.5, colors.HexColor('#cccccc')),
    ('PADDING',     (0,0), (-1,-1), 5),
    ('ALIGN',       (2,0), (2,-1), 'CENTER'),
]))
story.append(ok_table)
story.append(Spacer(1, 10))

# Locked table
story.append(Paragraph("SHOULD BE LOCKED (21 modules — expect upgrade prompt):", ParagraphStyle('RL', parent=body_style, textColor=RED, fontName='Helvetica-Bold', spaceAfter=4)))
locked = [
    "Customers", "Vehicles", "Appointments", "Job Cards", "Time Tracking",
    "Maintenance Schedules", "Inspections", "Estimates", "Repair Orders",
    "Technicians", "Parts", "Invoicing", "Payments", "Communication",
    "VIN Decode", "DTC Lookup", "Diagnostics", "AI Copilot",
    "Reports", "Labor Guide", "Login & Roles",
]
lock_rows = [["Module", "Expected Result", "Pass?"]]
for m in locked:
    lock_rows.append([m, "Shows upgrade gate / prompt", "[ ]"])
lock_table = Table(lock_rows, colWidths=[1.7*inch, 3.8*inch, 0.7*inch])
lock_table.setStyle(TableStyle([
    ('BACKGROUND',  (0,0), (-1,0), RED),
    ('TEXTCOLOR',   (0,0), (-1,0), colors.white),
    ('FONTNAME',    (0,0), (-1,0), 'Helvetica-Bold'),
    ('FONTSIZE',    (0,0), (-1,-1), 9),
    ('ROWBACKGROUNDS', (0,1), (-1,-1), [BGRED, colors.white]),
    ('GRID',        (0,0), (-1,-1), 0.5, colors.HexColor('#cccccc')),
    ('PADDING',     (0,0), (-1,-1), 5),
    ('ALIGN',       (2,0), (2,-1), 'CENTER'),
]))
story.append(lock_table)
story.append(Spacer(1, 6))
story.append(Paragraph("<b>Pass criteria:</b> All 21 modules show gate/upgrade prompt. Zero modules bypass the lock.", note_style))

# ── Step 5 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 5 — Verify Active Trial Still Has Full Access", h2_style))
story.append(Paragraph("<b>5a.</b> Open a second incognito window", body_style))
story.append(Paragraph("<b>5b.</b> Log in as your main owner account (active trial or pro)", body_style))
story.append(Paragraph("<b>5c.</b> Click every sidebar module — all should open normally", body_style))
story.append(Paragraph("<b>Pass criteria:</b> No lockouts on the active account. Every module accessible.", note_style))

# ── Step 6 ──────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=1, color=colors.HexColor('#dddddd'), spaceAfter=0))
story.append(Paragraph("Step 6 — Reset Test Account (Cleanup)", h2_style))
story.append(Paragraph("Restore the test account to an active trial for future test runs:", body_style))
story.append(Paragraph(
    "UPDATE public.profiles\n"
    "SET trial_ends_at = now() + INTERVAL '7 days'\n"
    "WHERE id = (\n"
    "  SELECT id FROM auth.users\n"
    "  WHERE email = 'testexpired@redlined1.com'\n"
    ");",
    code_style))

# ── Sign-off ────────────────────────────────────────────────────
story.append(HRFlowable(width="100%", thickness=2, color=RED, spaceBefore=18, spaceAfter=12))
story.append(Paragraph("Pass / Fail Sign-Off", h2_style))
signoff = [
    ["Tester Name:", "___________________________________", "Date Tested:", "________________"],
    ["Result:",      "[ ] PASS     [ ] FAIL",              "",              ""],
    ["Notes:",       "___________________________________________________________________", "", ""],
]
st = Table(signoff, colWidths=[1.1*inch, 2.8*inch, 1.1*inch, 1.2*inch])
st.setStyle(TableStyle([
    ('FONTSIZE',  (0,0), (-1,-1), 10),
    ('FONTNAME',  (0,0), (0,-1), 'Helvetica-Bold'),
    ('FONTNAME',  (2,0), (2,-1), 'Helvetica-Bold'),
    ('PADDING',   (0,0), (-1,-1), 5),
    ('VALIGN',    (0,0), (-1,-1), 'MIDDLE'),
]))
story.append(st)
story.append(Spacer(1, 8))
story.append(Paragraph(
    "If FAIL — document which modules were incorrectly accessible and file a bug report.",
    ParagraphStyle('warn', parent=note_style, textColor=RED)))

doc.build(story)
print(f"PDF written to {OUT}")

File: scripts/gen_expired_trial_test.js
const PDFDocument = require('pdfkit');
const fs = require('fs');
const path = require('path');

const OUT = path.join(__dirname, 'Test_Expired_Trial_Gate.pdf');
const doc = new PDFDocument({ margin: 55, size: 'LETTER' });
doc.pipe(fs.createWriteStream(OUT));

const RED    = '#cc0000';
const DARK   = '#111111';
const MUTED  = '#666666';
const GREEN  = '#1a7a1a';
const LGRAY  = '#dddddd';
const BGCODE = '#f5f5f5';

// ── helpers ──────────────────────────────────────────────────────
function rule(color = LGRAY, y) {
  const yy = y ?? doc.y;
  doc.moveTo(55, yy).lineTo(557, yy).strokeColor(color).lineWidth(color === RED ? 2 : 0.75).stroke();
  doc.moveDown(0.4);
}

function h1(text) {
  doc.fontSize(20).fillColor(RED).font('Helvetica-Bold').text(text, { continued: false });
}
function h2(text) {
  doc.moveDown(0.5);
  doc.fontSize(13).fillColor(DARK).font('Helvetica-Bold').text(text);
  doc.moveDown(0.25);
}
function body(text, indent = 0) {
  doc.fontSize(10).fillColor(DARK).font('Helvetica').text(text, 55 + indent, doc.y, { width: 502 - indent });
}
function muted(text) {
  doc.fontSize(9).fillColor(MUTED).font('Helvetica').text(text, { width: 502 });
}
function code(text) {
  const x = 55, w = 502, pad = 8;
  const lines = text.split('\n');
  const lineH = 13;
  const blockH = lines.length * lineH + pad * 2;
  doc.rect(x, doc.y, w, blockH).fill(BGCODE);
  const ty = doc.y + pad;
  doc.fontSize(8.5).fillColor(DARK).font('Courier').text(text, x + pad, ty, { width: w - pad * 2, lineGap: 2 });
  doc.moveDown(0.5);
}
function check(text, indent = 0) {
  doc.fontSize(10).fillColor(DARK).font('Helvetica')
    .text('☐  ' + text, 55 + indent, doc.y, { width: 502 - indent });
}

// row: [label, value, pass]
function tableRow(cols, widths, bg, isHeader) {
  const x = 55, pad = 6;
  const rowH = 20;
  let cx = x;
  if (bg) {
    doc.rect(x, doc.y, widths.reduce((a,b)=>a+b,0), rowH).fill(bg);
  }
  const ty = doc.y + 5;
  cols.forEach((col, i) => {
    doc.fontSize(9)
       .fillColor(isHeader ? '#ffffff' : DARK)
       .font(isHeader ? 'Helvetica-Bold' : 'Helvetica')
       .text(col, cx + pad, ty, { width: widths[i] - pad * 2, height: rowH });
    cx += widths[i];
  });
  doc.y += rowH;
}
function tableSection(headers, rows, headerBg, oddBg) {
  const w = [180, 262, 60];
  tableRow(headers, w, headerBg, true);
  rows.forEach((r, i) => tableRow(r, w, i % 2 === 0 ? oddBg : '#ffffff', false));
  // border
  const total = w.reduce((a,b)=>a+b,0);
  doc.rect(55, doc.y - rows.length * 20 - 20, total, rows.length * 20 + 20)
     .strokeColor(LGRAY).lineWidth(0.5).stroke();
  doc.moveDown(0.4);
}

// ── PAGE 1 ───────────────────────────────────────────────────────
h1('Expired Trial Plan Gate');
doc.fontSize(15).fillColor(DARK).font('Helvetica-Bold').text('Test Procedure');
doc.fontSize(10).fillColor(MUTED).font('Helvetica').text('Version 1.0  |  Date: 2026-06-17  |  Product: Redlined1');
doc.moveDown(0.5);
rule(RED);

// Objective
h2('Objective');
body('Verify that after a 7-day trial expires, the plan gate locks all modules except Dashboard, Settings, and Plans & Gates. Confirm that an active trial account retains unrestricted full access.');

// Pre-flight
h2('Pre-Flight Checklist');
check('You have access to the Supabase Dashboard SQL Editor');
check('You have a test email address ready (e.g. testexpired@redlined1.com)');
check('redlined1.com is deployed and accessible in a browser');
check('You are NOT logged in as your main owner account during these tests');
doc.moveDown(0.5);

rule();

// Step 1
h2('Step 1 — Create the Test Account  (one-time setup)');
body('1a.  Go to Supabase Dashboard → Authentication → Users → Invite User', 14);
body('1b.  Enter email: testexpired@redlined1.com  and a test password', 14);
body('1c.  Confirm the profile row exists — run in SQL Editor:', 14);
code("SELECT id, plan, trial_ends_at FROM public.profiles\nWHERE id = (SELECT id FROM auth.users\n            WHERE email = 'testexpired@redlined1.com');");
muted('Expected: row exists with plan = \'trial\' and trial_ends_at set ~7 days from signup.');
rule();

// Step 2
h2('Step 2 — Expire the Trial');
body('Run this SQL in Supabase → SQL Editor:');
code("UPDATE public.profiles\nSET plan = 'trial',\n    trial_ends_at = now() - INTERVAL '1 day'\nWHERE id = (\n  SELECT id FROM auth.users\n  WHERE email = 'testexpired@redlined1.com'\n);");
body('Then verify the update:');
code("SELECT u.email, p.plan, p.trial_ends_at\nFROM auth.users u\nJOIN public.profiles p ON p.id = u.id\nWHERE u.email = 'testexpired@redlined1.com';");
muted("Expected: trial_ends_at is in the past (yesterday's date or earlier).");
rule();

// Step 3
h2('Step 3 — Log In as the Expired Test Account');
body('3a.  Open a private / incognito browser window', 14);
body('3b.  Navigate to redlined1.com → Sign In', 14);
body('3c.  Enter credentials for testexpired@redlined1.com', 14);
rule();

// Step 4
h2('Step 4 — Verify Module Lock-Down');
body('Click every sidebar item and mark the result in the tables below.');
doc.moveDown(0.4);

body('SHOULD BE ACCESSIBLE (3 modules):', 0);
doc.fontSize(9).fillColor(GREEN).font('Helvetica-Bold').text('');
doc.moveDown(0.15);
tableSection(
  ['Module', 'Expected Result', 'Pass?'],
  [
    ['Dashboard',     'Opens normally — data is visible',   '[ ]'],
    ['Settings',      'Opens normally — can edit branding', '[ ]'],
    ['Plans & Gates', 'Opens upgrade / subscription page',  '[ ]'],
  ],
  GREEN, '#f0fff4'
);

body('SHOULD BE LOCKED (21 modules — expect upgrade prompt):', 0);
doc.moveDown(0.15);
tableSection(
  ['Module', 'Expected Result', 'Pass?'],
  [
    ['Customers',             'Shows upgrade gate / prompt', '[ ]'],
    ['Vehicles',              'Shows upgrade gate / prompt', '[ ]'],
    ['Appointments',          'Shows upgrade gate / prompt', '[ ]'],
    ['Job Cards',             'Shows upgrade gate / prompt', '[ ]'],
    ['Time Tracking',         'Shows upgrade gate / prompt', '[ ]'],
    ['Maintenance Schedules', 'Shows upgrade gate / prompt', '[ ]'],
    ['Inspections',           'Shows upgrade gate / prompt', '[ ]'],
    ['Estimates',             'Shows upgrade gate / prompt', '[ ]'],
    ['Repair Orders',         'Shows upgrade gate / prompt', '[ ]'],
    ['Technicians',           'Shows upgrade gate / prompt', '[ ]'],
    ['Parts',                 'Shows upgrade gate / prompt', '[ ]'],
    ['Invoicing',             'Shows upgrade gate / prompt', '[ ]'],
    ['Payments',              'Shows upgrade gate / prompt', '[ ]'],
    ['Communication',         'Shows upgrade gate / prompt', '[ ]'],
    ['VIN Decode',            'Shows upgrade gate / prompt', '[ ]'],
    ['DTC Lookup',            'Shows upgrade gate / prompt', '[ ]'],
    ['Diagnostics',           'Shows upgrade gate / prompt', '[ ]'],
    ['AI Copilot',            'Shows upgrade gate / prompt', '[ ]'],
    ['Reports',               'Shows upgrade gate / prompt', '[ ]'],
    ['Labor Guide',           'Shows upgrade gate / prompt', '[ ]'],
    ['Login & Roles',         'Shows upgrade gate / prompt', '[ ]'],
  ],
  RED, '#fff0f0'
);

muted('Pass criteria: All 21 modules above show gate/upgrade prompt. Zero modules bypass the lock.');
rule();

// Step 5
h2('Step 5 — Verify Active Trial Still Has Full Access');
body('5a.  Open a second incognito window', 14);
body('5b.  Log in as your main owner account (active trial or pro)', 14);
body('5c.  Click every sidebar module — all should open normally', 14);
doc.moveDown(0.25);
muted('Pass criteria: No lockouts on the active account. Every module is accessible.');
rule();

// Step 6
h2('Step 6 — Reset the Test Account  (Cleanup)');
body('Restore the test account to an active trial for future test runs:');
code("UPDATE public.profiles\nSET trial_ends_at = now() + INTERVAL '7 days'\nWHERE id = (\n  SELECT id FROM auth.users\n  WHERE email = 'testexpired@redlined1.com'\n);");

// Sign-off
rule(RED);
h2('Pass / Fail Sign-Off');
const soY = doc.y;
doc.fontSize(10).fillColor(DARK).font('Helvetica-Bold').text('Tester Name:', 55, soY);
doc.font('Helvetica').text('___________________________________', 140, soY);
doc.font('Helvetica-Bold').text('Date Tested:', 360, soY);
doc.font('Helvetica').text('________________', 440, soY);
doc.moveDown(0.8);
doc.font('Helvetica-Bold').text('Result:', 55, doc.y);
doc.font('Helvetica').text('☐  PASS        ☐  FAIL', 110, doc.y);
doc.moveDown(0.8);
doc.font('Helvetica-Bold').text('Notes:', 55, doc.y);
doc.font('Helvetica').text('________________________________________________________________________', 100, doc.y);
doc.moveDown(0.6);
doc.fontSize(9).fillColor(RED).font('Helvetica')
  .text('If FAIL — document which modules were incorrectly accessible and file a bug report.');

doc.end();
console.log('PDF written to', OUT);
